File: seller/src/main/java/com/example/seller/ui/home/HomeFragment.java
package com.example.seller.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.example.seller.Adapter.OrderAdapter;
import com.example.seller.Models.OrderModal;
import com.example.seller.SessionManager;
import com.example.seller.databinding.FragmentHomeBinding;
import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.firebase.database.FirebaseDatabase;

public class HomeFragment extends Fragment {

    OrderAdapter adapter;

    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        FragmentHomeBinding binding = FragmentHomeBinding.inflate(inflater, container, false);
        View root = binding.getRoot();
        final RecyclerView recyclerView = binding.orderRecyclerView;

        String phoneNo = new SessionManager(requireActivity()).getUserDetailFromSession().get(SessionManager.KEY_PHONE_NO);
        assert phoneNo != null;
        FirebaseRecyclerOptions<OrderModal> options = new FirebaseRecyclerOptions.Builder<OrderModal>()
                .setQuery(FirebaseDatabase.getInstance().getReference("Seller").child(phoneNo).child("Delivery order"), OrderModal.class)
                .build();
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, false));
        adapter = new OrderAdapter(options);
        recyclerView.setAdapter(adapter);

        return root;
    }


    @Override
    public void onStart() {
        super.onStart();
        adapter.startListening();
    }

    @Override
    public void onStop() {
        super.onStop();
        adapter.stopListening();
    }
}

File: app/src/main/java/com/example/loginApp/ui/home/HomeFragment.kt
package com.example.loginApp.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.loginApp.models.ShopModel
import com.example.loginApp.R
import com.example.loginApp.adapters.ShopAdapter
import com.firebase.ui.database.FirebaseRecyclerOptions
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.Query

class HomeFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_home, container, false)

        val recyclerView: RecyclerView = view.findViewById(R.id.shopRecyclerView)
        val query : Query = FirebaseDatabase.getInstance().getReference("Seller")
        val options = FirebaseRecyclerOptions.Builder<ShopModel>()
            .setQuery(query, ShopModel::class.java).build()
        val adapter = ShopAdapter(options)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(activity)
        adapter.startListening()
        return view
    }
}